lib/webtask.js

```diff
@@ -708,6 +765,36 @@ exports.ensure_authenticated = function(ctx) {
             });
         }
 
+        function validate_authentication_token(secret, expected_aud) {
+            var auth_token;
+            var decoded_auth_token;
+            var userinfo;
+
+            if(req.headers['authorization']) {
+                try {
+                    auth_token = req.headers['authorization'].match(/^bearer\s+(.+)$/i)[1];
+                } catch(e) {
+                    throw { code: 400, message: 'could not parse supplied Authorization header' + req.headers['authorization'] + ':' + e.message, req_id: req.req_id };
+                }
+            } else {
+                auth_token = req.query.key;
+            }
+
+            try {
+                userinfo = jsonwebtoken.verify(auth_token, secret);
```

Comment:
We already use `jws` in this file. Can this be used instead to validate the signature?